pkg/deploy/rp.go

```diff
@@ -498,7 +498,7 @@ func (g *generator) clustersKeyvault() *arm.Resource {
 			},
 			AccessPolicies: &[]mgmtkeyvault.AccessPolicyEntry{},
 		},
-		Name:     to.StringPtr("[concat(parameters('keyvaultPrefix'), '-clusters')]"),
+		Name:     to.StringPtr("[substring(concat(parameters('keyvaultPrefix'), '-clusters'), 0, 24)]"),
```

Comment:
might we use the parameter `maxLength` validator (https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-syntax#parameters) to achieve this (i.e. limit keyvaultPrefix to 24-max(len('-clusters'), len('-service')) bytes)?

it may also be worth shortening '-service', '-cluster' as well? wdyt?

Reply:
what about `-svc` and `-cls` ?

Reply:
sure